File: android-aop-core/src/main/java/com/flyjingfish/android_aop_core/annotations/IOThread.java
/*
 * Copyright (C) 2018 xuexiangjys(xuexiangjys@163.com)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.flyjingfish.android_aop_core.annotations;

import com.flyjingfish.android_aop_annotation.anno.AndroidAopPointCut;
import com.flyjingfish.android_aop_core.cut.IOThreadCut;
import com.flyjingfish.android_aop_core.enums.ThreadType;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * 切换到子线程的操作，加入此注解可使你的方法内的代码切换到子线程执行
 */
@AndroidAopPointCut(IOThreadCut.class)
@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.METHOD)
public @interface IOThread {

    /**
     * @return 子线程的类型，默认是多线程池
     */
    ThreadType value() default ThreadType.MultipleIO;
}

File: android-aop-core/src/main/java/com/flyjingfish/android_aop_core/annotations/IOThread.kt
package com.flyjingfish.android_aop_core.annotations

import com.flyjingfish.android_aop_annotation.anno.AndroidAopPointCut
import com.flyjingfish.android_aop_core.cut.IOThreadCut
import com.flyjingfish.android_aop_core.enums.ThreadType

/**
 * 切换到子线程的操作，加入此注解可使你的方法内的代码切换到子线程执行
 */
@AndroidAopPointCut(IOThreadCut::class)
@Retention(AnnotationRetention.RUNTIME)
@Target(
    AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER
)
annotation class IOThread(
    /**
     * 子线程的类型，默认是多线程池
     */
    val value: ThreadType = ThreadType.MultipleIO
)